pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -219,25 +250,81 @@ func (c *Controller) removeFinalizer(pvc *v1.PersistentVolumeClaim) error {
 	return nil
 }
 
-func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, error) {
+// This function checks if a PVC is in use and returns values inUse,
+// needsRequeue and error.
+// isUse: This value indicates if a PVC is in use by any pod or node
+// needsRequeue: This value indicates that the controller loop should
+// requeue the PVC in the controller, instead of waiting for a new event
+// handler call on the PVC. This is an optimization to react faster to a
+// change in state where no pod references the PVC.
+func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, bool, error) {
+	// If PVC is marked for delete, before the PVC.Spec.Volume is populated,
+	// we should not block PVC deletion. By the time PVC is deleted, if PV
+	// creation was already triggered and the volume plugin is in the process
+	// of creation of the volume, then after the PV is created, the PV controller
+	// will trigger a delete for the PV (if it has a reclaim policy of delete),
+	// since the claim will not be found.
+	var pv *v1.PersistentVolume
+	if pvc.Spec.VolumeName != "" {
+		var err error
+		pv, err = c.getPV(pvc)
+		if err != nil {
+			return false, false, err
+		}
+	}
+
 	// Look for a Pod using pvc in the Informer's cache. If one is found the
 	// correct decision to keep pvc is taken without doing an expensive live
-	// list.
-	if inUse, err := c.askInformer(pvc); err != nil {
+	// list. No requeue is needed as we will wait for a Pod event.
+	if inUse, err := c.askPodInformer(pvc); err != nil {
 		// No need to return because a live list will follow.
 		klog.Error(err)
 	} else if inUse {
-		return true, nil
+		return true, false, nil
```

Comment:
maybe add a comment why requeue is false?

Reply:
Done